src/packageurl/contrib/url2purl.py

```diff
@@ -53,7 +53,9 @@ def url2purl(url):
         try:
             return purl_router.process(url)
         except NoRouteAvailable:
-            return
+            # If `url` does not fit in one of the existing routes,
+            # we attempt to create a generic PackageURL for `url`
+            return build_generic_purl(url)
```

Comment:
@pombredanne I am not sure if this is the best place to create a generic purl for a given URL if there is no other route that handles the URL.

Reply:
I think this makes perfect sense to have this here :+1: